New Away Site: Hiskyn Treasure Trove

EJDenton opened the pull request:
This away site is intended to be a high risk, high reward, encounter. I'm also attempting to increase its replayability by incorporating a number of sections and submaps that will be newly generated each time. This is currently a work in progress but I've gotten a bit of the groundwork done and wanted to get eyes on it in case there were any objections to what's been done so far.

Things added or changed:
- Additional traps and environmental hazards.
- Lava is now more deadly. It used to only set you on fire, now it will burn you appropriately.
- Minor sprite fixes for uncommon elements for easier mapping; throwers, step triggers.
- Map and associated files, submap json, landmarks, areas, ect.

EJDenton commented:
!wip

kano-dot commented:
If the shuttles in the treasure room aren't meant to be functional, can you replace their consoles with `terminal/inactive` subtype? Also if they're scrapped aside it'd be fitting to have them depowered and dark, their SMES extracted/drained

---

<img width="399" height="329" alt="Screenshot_24" src="https://github.com/user-attachments/assets/64f3aa7b-0948-4faf-8437-cc7a53f762b3" />

Some leftover dirt decals here on the chasm (187, 199) and the bit 8 tiles to the right from here

---

<img width="743" height="319" alt="Screenshot_26" src="https://github.com/user-attachments/assets/a15a81f1-97d0-42e2-8c67-c917ee2014fa" />

Decals like this will appear underneath the grates. You could perhaps move them to the middle part or remove the grate they're standing under?

---

<img width="227" height="387" alt="Screenshot_27" src="https://github.com/user-attachments/assets/5efc6a86-ee21-4f00-8f76-63d47fab7aac" />

Please align the blastdoors in this side

---

<img width="373" height="599" alt="Screenshot_28" src="https://github.com/user-attachments/assets/4cb179f4-99a9-4eb8-9474-4d007ed647fe" />

Something I noticed, `.../exoplanet/asteroid/basalt` type has its icon offset, probably trying to smooth with `.../exoplanet/basalt/cave` turfs and rocks in its border for some reason. This is a bug not related to this PR however and could be easily fixed afterwards!

---

<img width="686" height="359" alt="Screenshot_29" src="https://github.com/user-attachments/assets/e8ebfccb-d0e7-44c2-9a28-30a2ef527836" />

This is me being a bit nitpicky, we have platform corners like this that can smoothen the overlapping parts. Incase this is a style you'd like to go for

kano-dot commented on maps/away/away_site/treasure_trove/treasure_trove_landmarks.dm:
```diff
@@ -0,0 +1,66 @@
+/obj/effect/shuttle_landmark/trove/landing_01
+	name = "Beach Landing Zone 01"
+	landmark_tag = "nav_point_beach_01"
+	dir = 1
+
+/obj/effect/shuttle_landmark/trove/landing_02a
+	name = "Beach Landing Zone 02a"
+	landmark_tag = "nav_point_beach_02a"
+	dir = 2
+
+/obj/effect/shuttle_landmark/trove/landing_02b
+	name = "Beach Landing Zone 02b"
+	landmark_tag = "nav_point_beach_02b"
+	dir = 4
+
+/obj/effect/shuttle_landmark/trove/landing_02c
+	name = "Beach Landing Zone 02c"
+	landmark_tag = "nav_point_beach_02c"
+	dir = 1
+
+/obj/effect/shuttle_landmark/trove/landing_02d
+	name = "Beach Landing Zone 02d"
+	landmark_tag = "nav_point_beach_02d"
+	dir = 8
+
+/obj/effect/shuttle_landmark/trove/landing_03
+	name = "Beach Landing Zone 03"
+	landmark_tag = "nav_point_beach_03"
+	dir = 4
+
+/obj/effect/shuttle_landmark/trove/landing_04
+	name = "Beach Landing Zone 04"
+	landmark_tag = "nav_point_beach_04"
+	dir = 8
```
```suggestion
	dir = NORTH

/obj/effect/shuttle_landmark/trove/landing_02a
	name = "Beach Landing Zone 02a"
	landmark_tag = "nav_point_beach_02a"
	dir = SOUTH

/obj/effect/shuttle_landmark/trove/landing_02b
	name = "Beach Landing Zone 02b"
	landmark_tag = "nav_point_beach_02b"
	dir = EAST

/obj/effect/shuttle_landmark/trove/landing_02c
	name = "Beach Landing Zone 02c"
	landmark_tag = "nav_point_beach_02c"
	dir = NORTH

/obj/effect/shuttle_landmark/trove/landing_02d
	name = "Beach Landing Zone 02d"
	landmark_tag = "nav_point_beach_02d"
	dir = WEST

/obj/effect/shuttle_landmark/trove/landing_03
	name = "Beach Landing Zone 03"
	landmark_tag = "nav_point_beach_03"
	dir = EAST

/obj/effect/shuttle_landmark/trove/landing_04
	name = "Beach Landing Zone 04"
	landmark_tag = "nav_point_beach_04"
	dir = WEST
```

kano-dot commented on maps/away/away_site/treasure_trove/treasure_trove.dm:
```diff
@@ -0,0 +1,94 @@
+/obj/item/paper/fluff/treasure_trove/survivor
+	name = "journal page"
+	desc = "A single page remaining from a survivor's journal."
+	info = "\
+		I'm going to die here. This place is a death trap.<br>\
+		We buried Rena, Markus, and the unathi but pretty quick something came<br>\
+		and started digging them up. Staying away from the jungle.<br>\
+		<br>\
+		Alsir and Jena are intent on going into the caves. I told them no.\
+		I'm not looking to die but it seems like this place is\
+		hungry for blood anyway.\
+		<br>\
+		They went into the caves two days ago.\
+		They haven't come back.\
+		<br>\
+		Nearly got carried away by a riptide. Even the water is trying to\
+		kill me.\
+		<br>\
+		Found what seems to be an old prospecting setup, but\
+		it's too close to the jungle. I think something\
+		caught my scent. It always feels like something is\
+		watching me... waiting.\
+		"
```
```suggestion
	info = "\
		I'm going to die here. This place is a death trap.<br> \
		We buried Rena, Markus, and the unathi but pretty quick something came<br> \
		and started digging them up. Staying away from the jungle.<br> \
		<br> \
		Alsir and Jena are intent on going into the caves. I told them no. \
		I'm not looking to die but it seems like this place is hungry for blood anyway. \
		<br> \
		They went into the caves two days ago. \
		They haven't come back. \
		<br> \
		Nearly got carried away by a riptide. Even the water is trying to kill me. \
		<br> \
		Found what seems to be an old prospecting setup, but it's too close to the jungle. I think something \
		caught my scent. It always feels like something is watching me... waiting. \
		"
```

EJDenton commented:
> If the shuttles in the treasure room aren't meant to be functional, can you replace their consoles with `terminal/inactive` subtype? Also if they're scrapped aside it'd be fitting to have them depowered and dark, their SMES extracted/drained
> <img alt="Screenshot_24" width="399" height="329" src="https://private-user-images.githubusercontent.com/89972582/491908396-64f3aa7b-0948-4faf-8437-cc7a53f762b3.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTgzNzcyOTUsIm5iZiI6MTc1ODM3Njk5NSwicGF0aCI6Ii84OTk3MjU4Mi80OTE5MDgzOTYtNjRmM2FhN2ItMDk0OC00ZmFmLTg0MzctY2M3YTUzZjc2MmIzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA5MjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwOTIwVDE0MDMxNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTgxODlmNzA1MjdiOTc4MmQ2Nzg5ZDg4YzQ2MWMzOTFjY2Q1ZmI2ODRlN2ZmZmJhODk3Y2ZiNWM4NTUyZmJjNmUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.rhK4BeEifoFrv3uiDLfWHZl6MY7JHCvUxN6xEDDci88">
> 
> Some leftover dirt decals here on the chasm (187, 199) and the bit 8 tiles to the right from here
> <img alt="Screenshot_26" width="743" height="319" src="https://private-user-images.githubusercontent.com/89972582/491908713-a15a81f1-97d0-42e2-8c67-c917ee2014fa.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTgzNzcyOTUsIm5iZiI6MTc1ODM3Njk5NSwicGF0aCI6Ii84OTk3MjU4Mi80OTE5MDg3MTMtYTE1YTgxZjEtOTdkMC00MmUyLThjNjctYzkxN2VlMjAxNGZhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA5MjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwOTIwVDE0MDMxNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZiYzNhNDUzMGIxYjdmMTNhYmQ4YmM2MDBhMjlkMTgyOWNjMThjMjM1OTc2YjE3ZWEyYzJhNzczYWQ4ODExMmUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.p9aBiom_ssUA8HiYkS_P11nKNBTTvEZUf3OG6t9hoZA">
> 
> Decals like this will appear underneath the grates. You could perhaps move them to the middle part or remove the grate they're standing under?
> <img alt="Screenshot_27" width="227" height="387" src="https://private-user-images.githubusercontent.com/89972582/491908990-5efc6a86-ee21-4f00-8f76-63d47fab7aac.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTgzNzcyOTUsIm5iZiI6MTc1ODM3Njk5NSwicGF0aCI6Ii84OTk3MjU4Mi80OTE5MDg5OTAtNWVmYzZhODYtZWUyMS00ZjAwLThmNzYtNjNkNDdmYWI3YWFjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA5MjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwOTIwVDE0MDMxNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWUzODU1YjYxNWEyNDg2ZWEwMWFmNzlmYWIwYzM3Yzc4M2IxMTAyOWM2OWE2MDAyZmE5NThkN2VkY2UzNjUxM2EmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.VzhTnwP5IGPHelYw_0LfzW3ma7R8hU_7LpMU1ymcxx4">
> 
> Please align the blastdoors in this side
> <img alt="Screenshot_28" width="373" height="599" src="https://private-user-images.githubusercontent.com/89972582/491909066-4cb179f4-99a9-4eb8-9474-4d007ed647fe.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTgzNzcyOTUsIm5iZiI6MTc1ODM3Njk5NSwicGF0aCI6Ii84OTk3MjU4Mi80OTE5MDkwNjYtNGNiMTc5ZjQtOTlhOS00ZWI4LTk0NzQtNGQwMDdlZDY0N2ZlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA5MjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwOTIwVDE0MDMxNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRkOTJjYzUwMmY3NDljZDFhYzc3Y2E3M2NmODhhMzdjZDBlODMwNDE4ZjI5YzFiOTNjZmUxZDUyYjZiOWNhYTImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.3u14vt36v0_shP4q8Oahu--I6CdOMtkj9wRYuO7H2Qc">
> 
> Something I noticed, `.../exoplanet/asteroid/basalt` type has its icon offset, probably trying to smooth with `.../exoplanet/basalt/cave` turfs and rocks in its border for some reason. This is a bug not related to this PR however and could be easily fixed afterwards! Though if you can fix it or replace those it'd be also feasible
> <img alt="Screenshot_29" width="686" height="359" src="https://private-user-images.githubusercontent.com/89972582/491909588-e8ebfccb-d0e7-44c2-9a28-30a2ef527836.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTgzNzcyOTUsIm5iZiI6MTc1ODM3Njk5NSwicGF0aCI6Ii84OTk3MjU4Mi80OTE5MDk1ODgtZThlYmZjY2ItZDBlNy00NGMyLTlhMjgtMzBhMmVmNTI3ODM2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA5MjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwOTIwVDE0MDMxNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY0NzA4OGU0YjMzN2VlMmI5MTI1Mjk5NGRhOTJkZWVmZjljMTA3NjdmYWY0ZTBmZjI0N2IxNTQ2MWRiZGE0MzQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.3m5Hi6uWQ7kLd9tSN8gGvijOblp_W6ED7Uv7ZQGBn70">
> 
> This is me being a bit nitpicky, we have platform corners like this that can smoothen the overlapping parts. Incase this is a style you'd like to go for

I applied all the changes requested with some standouts. The exoplanet turf offset was something I noticed as well, and I tried to map in such a way as it's less noticeable. I want to keep /exoplanet/asteroid/basalt in the lower area since it renders with very nice random lava elements and looks less flat but the area you highlighted had some errant turf from my copy +pasting. I've corrected it so it looks more uniform.

<img width="304" height="619" alt="image" src="https://github.com/user-attachments/assets/2a6cf977-4b8a-4f54-9812-88e05a87b353" />

I also made changes to the general turf in the lower zlevel as I originally used exoplanet turf that was lighting incorrectly. I altered the shuttles and removed some lights but I want them to be semi-lit just to help with general visibility.

EJDenton commented on code/game/objects/effects/step_triggers.dm:
```diff
@@ -118,6 +118,9 @@
 	///List of moving atoms mapped to their inital direction
 	var/list/affecting = list()
 
+	icon = 'icons/mob/screen/generic.dmi'
+	icon_state = "dir_arrow"
```
Change applied.

EJDenton commented on code/game/machinery/portable_turret.dm:
```diff
@@ -1208,6 +1208,32 @@
+	no_salvage = TRUE
+	req_one_access = list(ACCESS_SYNDICATE)
+
+/obj/machinery/porta_turret/hologram/assembly_hints(mob/user, distance, is_adjacent)
+	. = ..()
+	. = null
+
+/obj/machinery/porta_turret/hologram/condition_hints(mob/user, distance, is_adjacent)
+	. = ..()
+	. = null
```
After additional testing it looks like this method isn't working properly to remove the condition and assembly hints so I've removed it entirely.
